src/controllers/hid/hidreportdescriptor.h

```diff
@@ -0,0 +1,258 @@
+    LongItemKeyword   = 0b1111'11'10,
+
+    AllSizeBitsMask   = 0b0000'00'11
+};
+
+// Collection types according to chapter 6.2.2.6 of HID class definition 1.11
+enum class CollectionType : uint8_t {
+    Physical      = 0x00, // e.g. group of axes
+    Application   = 0x01, // e.g. mouse or keyboard
+    Logical       = 0x02, // interrelated data
+    Report        = 0x03,
+    NamedArray    = 0x04,
+    UsageSwitch   = 0x05,
+    UsageModifier = 0x06,
+    Reserved      = 0x07, // range of 0x07-0x7F
+    VendorDefined = 0x80  // range of 0x80-0xFF
+};
+// clang-format on
+
+union ControlFlags {
+    struct {
+        uint32_t data_constant : 1;          // Data (0) | Constant (1)
+        uint32_t array_variable : 1;         // Array (0) | Variable (1)
+        uint32_t absolute_relative : 1;      // Absolute (0) | Relative (1)
+        uint32_t no_wrap_wrap : 1;           // No Wrap (0) | Wrap (1)
+        uint32_t linear_non_linear : 1;      // Linear (0) | Non Linear (1)
+        uint32_t preferred_no_preferred : 1; // Preferred State (0) | No Preferred (1)
+        uint32_t no_null_null : 1;           // No Null position (0) | Null state(1)
+        uint32_t non_volatile_volatile : 1;  // Non Volatile (0) | Volatile (1)
+        uint32_t bit_field_buffered : 1;     // Bit Field (0) | Buffered Bytes (1)
+        uint32_t reserved : 23;
+    };
+    uint32_t payload;
+};
+
+// Class representing a control described in the HID report descriptor
+class Control {
+  public:
+    Control(const ControlFlags flags,
+            const uint32_t usage,
+            const int32_t logicalMinimum,
+            const int32_t logicalMaximum,
+            const int32_t physicalMinimum,
+            const int32_t physicalMaximum,
+            const int8_t unitExponent,
+            const uint32_t unit,
+            const uint16_t bytePosition, // Position of the first byte in the report
+            const uint8_t bitPosition,   // Position of first bit in first byte
+            const uint8_t bitSize);
+
+    const ControlFlags m_flags;
+
+    const uint32_t m_usage;
+    const int32_t m_logicalMinimum;
+    const int32_t m_logicalMaximum;
+    const int32_t m_physicalMinimum;
+    const int32_t m_physicalMaximum;
+    const int8_t m_unitExponent;
+    const uint32_t m_unit;
+    const uint16_t m_bytePosition; // Position of the first byte in the report
+    const uint8_t m_bitPosition;   // Position of first bit in first byte
+    const uint8_t m_bitSize;
+
+  private:
+};
+
+int32_t extractLogicalValue(const QByteArray& data, const Control& control);
+bool applyLogicalValue(QByteArray& data, const Control& control, int32_t controlValue);
+
+// Class representing a report in the HID report descriptor
+class Report {
+  public:
+    Report(const HidReportType& reportType, const uint8_t& reportId);
+
+    void addControl(const Control& item);
+    void increasePosition(unsigned int bitSize);
+    std::pair<uint16_t, uint8_t> getLastPosition() const {
```

Comment:
Can we jsut introduce getLastBytePosition() and  getLastBitPosition(), for a more easy to use interface?